tests/models/opt/test_modeling_opt.py

```diff
@@ -268,17 +268,22 @@ def _long_tensor(tok_lst):
 @require_torch
 class OPTModelIntegrationTests(unittest.TestCase):
     @slow
+    @unittest.skipIf(torch_device != "cpu", "Cannot make deterministic on GPU")
```

Comment:
Getting different logits results on GPU depending on PyTorch version (1.10+cu11.0 vs. 1.11+cu11.4) and results also differ between CPU and GPU. Only on CPU it seems to be deterministic.

It's not because the weights are saved & loaded in FP16 - checked that the same happens when weights are stored in fp32 and loaded in fp32. 
It's across multiple checkpoints (not just 350m, but also 125m).
The differences start to creep in in the first linear projection matrix `project_in_dim` for the 350m model and the first attention layer for 125m.

Reply:
Upgrading precision to 1e-2 so that tests pass on GPU - think that's the best we can do

Reply:
**Interesting observation:**

- BART has the same problem, but the differences are not as flagrant in the end
- BERT does **not** have this problem at all. Results are exactly the same.

cc @stephenroller @stas00 @sgugger @LysandreJik

Reply:
getting logits to exact match is very difficult. Perhaps a much longer input and checking that softmax output matches - after all logits fluctuations don't matter at the end but 100% softmax match does.

Reply:
if you use a very high tolerance then it becomes questionable whether the test is doing anything.